usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -0,0 +1,1523 @@
+
+        final SortedMap<Integer, StopTime> mockThirdTripStopTimeCollection = new TreeMap<>();
+        final StopTime thirdTripFirstStopTime = mock(StopTime.class);
+        when(thirdTripFirstStopTime.getArrivalTime()).thenReturn(940);
+        when(thirdTripFirstStopTime.getDepartureTime()).thenReturn(1216);
+        when(thirdTripFirstStopTime.getStopSequence()).thenReturn(14);
+
+        final StopTime thirdTripLastStopTime = mock(StopTime.class);
+        when(thirdTripLastStopTime.getArrivalTime()).thenReturn(1308);
+        when(thirdTripLastStopTime.getDepartureTime()).thenReturn(1350);
+        when(thirdTripLastStopTime.getStopSequence()).thenReturn(15);
+
+        mockThirdTripStopTimeCollection.put(14, thirdTripFirstStopTime);
+        mockThirdTripStopTimeCollection.put(15, thirdTripLastStopTime);
+
+        when(mockDataRepo.getAllTripByBlockId()).thenReturn(mockTripPerBlockIdCollection);
+        when(mockDataRepo.getStopTimeByTripId("2")).thenReturn(mockFirstTripStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("5")).thenReturn(mockSecondTripStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("8")).thenReturn(mockThirdTripStopTimeCollection);
+
+        final Map<String, Calendar> mockCalendarCollection = new HashMap<>();
+        final Calendar calendarForServiceA = mock(Calendar.class);
+        final Calendar calendarForServiceB = mock(Calendar.class);
+        final Calendar calendarForServiceC = mock(Calendar.class);
+        mockCalendarCollection.put("a", calendarForServiceA);
+        mockCalendarCollection.put("b", calendarForServiceB);
+        mockCalendarCollection.put("c", calendarForServiceC);
+        when(calendarForServiceA.isOverlapping(calendarForServiceB)).thenReturn(false);
+        when(calendarForServiceA.isOverlapping(calendarForServiceC)).thenReturn(true);
+        when(calendarForServiceB.isOverlapping(calendarForServiceC)).thenReturn(false);
+        final LocalDate serviceAStartDate = LocalDate.of(2020, 8, 1);
+        final LocalDate serviceAEndDate = LocalDate.of(2020, 8, 31);
+        final LocalDate serviceBStartDate = LocalDate.of(2020, 9, 1);
+        final LocalDate serviceBEndDate = LocalDate.of(2020, 9, 30);
+        final LocalDate serviceCStartDate = LocalDate.of(2020, 8, 1);
+        final LocalDate serviceCEndDate = LocalDate.of(2020, 8, 31);
+
+        when(calendarForServiceA.getStartDate()).thenReturn(serviceAStartDate);
+        when(calendarForServiceA.getEndDate()).thenReturn(serviceAEndDate);
+        when(calendarForServiceB.getStartDate()).thenReturn(serviceBStartDate);
+        when(calendarForServiceB.getEndDate()).thenReturn(serviceBEndDate);
+        when(calendarForServiceC.getStartDate()).thenReturn(serviceCStartDate);
+        when(calendarForServiceC.getEndDate()).thenReturn(serviceCEndDate);
+        when(calendarForServiceA.getOverlappingDays(calendarForServiceC))
```

Comment:
Same on not mocking domain class methods

Reply:
As discussed, implementation of domain layer classes is tested separately, so that tests of use case layer only verify interactions with mocked objects.